docs/multicast-guide.md

```diff
@@ -0,0 +1,122 @@
+# Multicast User Guide
+
+Antrea supports Multicast traffic in the following scenarios:
+
+1. Pods to Pods, any Pod that has joined the multicast group should receive the traffic from the Pod senders send multicast traffic to that group.
+2. Pods to External, external hosts connected to the Node will receive the multicast traffic from that local Pods.
+3. External to Pod, multicast traffic sent by external hosts connected to the Nodes will be received if some local Pods have joined the multicast group.
+4. Mixture of cases from 1,2,3.
+
+## Table of Contents
+
+<!-- toc -->
```

Comment:
Please run `make verify` locally and fix the issue.

Reply:
updated